Documentation/contributing/testing/e2e_legacy.rst

```diff
@@ -0,0 +1,739 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _testsuite-legacy:
+
+End-To-End Testing Framework (Legacy)
+=====================================
+
+.. note::
+
+   Cilium is in a process of migrating from Ginkgo to a new framework for
+   writing and running end-to-end tests. Please refer to
+   :ref:`the new e2e documentation<testsuite>`.
+
+Introduction
+~~~~~~~~~~~~
+
+Cilium uses `Ginkgo`_ as a testing framework for
+writing end-to-end tests which test Cilium all the way from the API level (e.g.
+importing policies, CLI) to the datapath (i.e, whether policy that is imported
+is enforced accordingly in the datapath).  The tests in the ``test`` directory
+are built on top of Ginkgo. Ginkgo provides a rich framework for developing
+tests alongside the benefits of Golang (compilation-time checks, types, etc.).
+To get accustomed to the basics of Ginkgo, we recommend reading the `Ginkgo
+Getting-Started Guide
+<https://onsi.github.io/ginkgo/#getting-started-writing-your-first-test>`_ , as
+well as running `example tests
+<https://github.com/onsi/composition-ginkgo-example>`_ to get a feel for the
+Ginkgo workflow.
+
+These test scripts will invoke ``vagrant`` to create virtual machine(s) to
+run the tests. The tests make heavy use of the Ginkgo `focus`_ concept to
+determine which VMs are necessary to run particular tests. All test names
+*must* begin with one of the following prefixes:
+
+* ``Runtime``: Test cilium in a runtime environment running on a single node.
+* ``K8s``: Create a small multi-node kubernetes environment for testing
+  features beyond a single host, and for testing kubernetes-specific features.
+
+.. _Ginkgo: https://onsi.github.io/ginkgo/
+.. _focus: `Focused Specs`_
+
+Running End-To-End Tests
+~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
This section appears to be missing content. If there won't be any for this PR, remove the heading instead of leaving it unpopulated.